libraries/eosiolib/core/eosio/crypto_bls_ext.hpp

```diff
@@ -549,59 +470,35 @@ namespace detail {
     }
 }
 
-    inline std::string encode_g1_to_bls_public_key(const bls_g1_affine& g1) {
-        return detail::bls_type_to_base64<bls_g1_affine, detail::bls_public_key_prefix>(g1);
+    inline std::string encode_g1_to_bls_public_key(const bls_g1& g1) {
+        return detail::bls_type_to_base64<bls_g1, detail::bls_public_key_prefix>(g1);
     }
-    inline bls_g1_affine decode_bls_public_key_to_g1(std::string_view public_key) {
-        return detail::bls_base64_to_type<bls_g1_affine, detail::bls_public_key_prefix>(public_key.data(), public_key.size());
+    inline bls_g1 decode_bls_public_key_to_g1(std::string_view public_key) {
+        return detail::bls_base64_to_type<bls_g1, detail::bls_public_key_prefix>(public_key.data(), public_key.size());
     }
-    inline std::string encode_g2_to_bls_signature(const bls_g2_affine& g2) {
-        return detail::bls_type_to_base64<bls_g2_affine, detail::bls_signature_prefix>(g2);
+    inline std::string encode_g2_to_bls_signature(const bls_g2& g2) {
+        return detail::bls_type_to_base64<bls_g2, detail::bls_signature_prefix>(g2);
     }
-    inline bls_g2_affine decode_bls_signature_to_g2(std::string_view public_key) {
-        return detail::bls_base64_to_type<bls_g2_affine, detail::bls_signature_prefix>(public_key.data(), public_key.size());
-    }
-
-    inline bls_g1 g1_affine_to_jacobian(const bls_g1_affine& pubkey) {
-        using namespace detail;
-
-        static_assert(sizeof(bls_g1_affine) + sizeof(R1) == sizeof(bls_g1));
-        // add z coordinate (R1) to pubkey and signature_proof
-        bls_g1 pubkey_jacobian;
-        auto insert_it = std::copy(pubkey.begin(), pubkey.end(), pubkey_jacobian.begin());
-        std::copy(R1.begin(), R1.end(), insert_it);
-
-        return pubkey_jacobian;
-    }
-
-    inline bls_g2 g2_affine_to_jacobian(const bls_g2_affine& signature_proof) {
-        using namespace detail;
-        
-        bls_g2 sig_ex = {0};
-        auto insert_it = std::copy(signature_proof.begin(), signature_proof.end(), sig_ex.begin());
-        std::copy(R1.begin(), R1.end(), insert_it);
-
-        return sig_ex;
+    inline bls_g2 decode_bls_signature_to_g2(std::string_view public_key) {
+        return detail::bls_base64_to_type<bls_g2, detail::bls_signature_prefix>(public_key.data(), public_key.size());
     }
 
     // pubkey and signature are assumed to be in RAW affine little-endian bytes
-    bool bls_pop_verify(const bls_g1_affine& pubkey, const bls_g2_affine& signature_proof) {
+    bool bls_pop_verify(const bls_g1& pubkey, const bls_g2& signature_proof) {
         using namespace detail;
 
         bls_g1 g1_points[2] = {0};
         bls_g2 g2_points[2] = {0};
 
-        bls_g1 pubkey_ex = g1_affine_to_jacobian(pubkey);
-        bls_g2 sig_ex = g2_affine_to_jacobian(signature_proof);
+        memcpy(g1_points[0].data(), G1_ONE_NEG.data(), G1_ONE_NEG.size());
```

Comment:
Do you want to use `std::memcpy`?

Would be it safer to use `g1_points[0].size()` for copying size?

Reply:
Done